cmd/kubectl/kubectlcobra/grouping.go

```diff
@@ -57,3 +60,82 @@ func sortGroupingObject(infos []*resource.Info) bool {
 	}
 	return false
 }
+
+// Adds the inventory of all objects (passed as infos) to the
+// grouping object. Returns an error if a grouping object does not
+// exist, or we are unable to successfully add the inventory to
+// the grouping object; nil otherwise. Each object is in
+// unstructured.Unstructured format.
+func addInventoryToGroupingObj(infos []*resource.Info) error {
+
+	// Iterate through the objects (infos), creating an Inventory struct
+	// as metadata for the object, or if it's the grouping object, store it.
+	var groupingObj *unstructured.Unstructured
+	inventoryMap := map[string]string{}
+	for _, info := range infos {
+		obj := info.Object
+		if isGroupingObject(obj) {
+			var ok bool
+			groupingObj, ok = obj.(*unstructured.Unstructured)
```

Comment:
if we find more than one we retain only the last one?

Reply:
More than one grouping object should be an error. I will update the code.